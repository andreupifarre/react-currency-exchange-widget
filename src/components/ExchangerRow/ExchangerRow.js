import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import CurrencyDropdown from '../CurrencyDropdown/CurrencyDropdown';
import './ExchangerRow.scss';

const ExchangerRow = ({
  alt,
  balance,
  currency,
  sign,
  amount,
  info,
  currencies,
  onCurrencyChange,
  onAmountChange,
  autoFocus,
}) => {
  const handleChange = (e) => {
    onAmountChange(e.target.value);
  };

  return (
    <div className={classNames('exchanger-row', { 'exchanger-row--alt': alt })}>
      <div className="exchanger-row__line">
        <CurrencyDropdown
          currency={currency}
          currencies={currencies}
          onCurrencyChange={onCurrencyChange}
        />
        <input
          type="number"
          className="exchanger-row__amount"
          autoComplete="off"
          placeholder="0"
          onChange={handleChange}
          value={amount}
          autoFocus={autoFocus}
        />
      </div>
      <div className="exchanger-row__line">
        <span className="exchanger-row__balance">
          Balance: {sign}
          {balance}
        </span>
        <span className="exchanger-row__fee">{info}</span>
      </div>
    </div>
  );
};

export default ExchangerRow;

ExchangerRow.defaultProps = {
  alt: false,
  balance: '',
  currency: '',
  sign: '',
  amount: '',
  info: '',
  currencies: [],
  onCurrencyChange: () => {},
  onAmountChange: () => {},
  autoFocus: false,
};

ExchangerRow.propTypes = {
  alt: PropTypes.bool.isRequired,
  balance: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  sign: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  info: PropTypes.string.isRequired,
  currencies: PropTypes.array.isRequired,
  onCurrencyChange: PropTypes.func.isRequired,
  onAmountChange: PropTypes.func.isRequired,
  autoFocus: PropTypes.bool,
};
