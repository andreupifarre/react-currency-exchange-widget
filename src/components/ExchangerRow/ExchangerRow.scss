.exchanger-row {
  margin: 0 -34px;
  padding: 60px 34px;

  &--alt {
    background-color: white;
  }

  &__line {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    min-height: 38px;
  }

  &__balance {
    font-size: 17px;
    margin-right: 20px;
    color: #aaa;
  }

  &__fee {
    font-size: 17px;
    flex: 1;
    text-align: right;
    color: #aaa;
  }

  &__amount {
    background-color: transparent;
    border: 0;
    font-size: 22px;
    text-align: right;
    flex: 1;
    outline: none;

    &::placeholder {
      color: #aaa;
    }
  }
}
