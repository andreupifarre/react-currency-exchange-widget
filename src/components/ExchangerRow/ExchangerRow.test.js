import React from 'react';
import { shallow } from 'enzyme';

import ExchangerRow from './ExchangerRow';

const render = (props) => shallow(<ExchangerRow {...props} />);

describe('ExchangerRow', () => {
  it('should contain a parent element with a class .exchanger-row', () => {
    const element = render().find('.exchanger-row');
    expect(element.exists()).toBeTruthy();
  });

  it('should contain a CurrencyDropdown element', () => {
    const element = render().find('CurrencyDropdown');
    expect(element.exists()).toBeTruthy();
  });

  it('should contain an input field with a class .exchanger-row__amount', () => {
    const element = render().find('.exchanger-row__amount');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('input');
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
