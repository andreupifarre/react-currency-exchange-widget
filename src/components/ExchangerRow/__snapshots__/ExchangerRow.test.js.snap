// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangerRow should render correctly 1`] = `
Array [
  <div
    className="exchanger-row"
  >
    <div
      className="exchanger-row__line"
    >
      <CurrencyDropdown
        currencies={Array []}
        currency=""
        onCurrencyChange={[Function]}
      />
      <input
        autoComplete="off"
        autoFocus={false}
        className="exchanger-row__amount"
        onChange={[Function]}
        placeholder="0"
        type="number"
        value=""
      />
    </div>
    <div
      className="exchanger-row__line"
    >
      <span
        className="exchanger-row__balance"
      >
        Balance: 
        
        
      </span>
      <span
        className="exchanger-row__fee"
      >
        
      </span>
    </div>
  </div>,
]
`;
